<launch>
    <!-- arguments of the tello node -->
    <arg name="namespace" default="tello" />
    <arg name="tello_ip" default="192.168.10.1" />
    <arg name="tello_cmd_server_port" default="8889" />
    <arg name="local_cmd_client_port" default="8890" />
    <arg name="local_vid_server_port" default="6038" />
    <arg name="camera_calibration" default="$(find tello_node)/config/960x720.yaml" />

    <group ns="$(arg namespace)">
        <node pkg="tello_node" type="tello_driver_node.py" name="tello_driver_node" output="screen">
            <param name="tello_ip" value="$(arg tello_ip)" />
            <param name="tello_cmd_server_port" value="$(arg tello_cmd_server_port)" />
            <param name="local_cmd_client_port" value="$(arg local_cmd_client_port)" />
            <param name="local_vid_server_port" value="$(arg local_vid_server_port)" />
            <param name="initial_frame" value="100" />
            <param name="connect_time_out" value="10.0" />
            <param name="h264_encoded_stream" value="true" />
            <param name="camera_calibration" value="$(arg camera_calibration)" />
            <remap from="/tello/camera_info" to="/camera/camera_info" />
        </node>
        
        <!-- image image_transport node to convert the compressed image into raw image for ORB SLAM2 -->
        <node pkg="image_transport" name="image_uncompressed" type="republish" args="h264 in:=/tello/image_raw raw out:=/tello/image_raw" />
        <node pkg="tf2_ros" name="camera_offset" type="static_transform_publisher" args="0 0 0 0 0 0 base_link camera_link" />
    </group>

    <!-- <node name="rqt_image_view" pkg="rqt_image_view" type="rqt_image_view" respawn="false" output="screen" /> -->
    <!-- <include file="$(find keyboard_teleop)/launch/keyboard_teleop.launch" /> -->
    <include file="/home/koko/orbslam2_ws/src/orb_slam_2_ros/ros/launch/orb_slam2_tello_mono.launch" />
    <include file="$(find controller_interface)/launch/rviz.launch" />
    <!-- <include file="$(find keyboard_teleop)/launch/keyboard_teleop.launch" /> -->
    <!-- <include file="/home/koko/orb_feature_ws/src/orbslam-map-saving-extension/orb_slam2_ros/launch/orb_test.launch" /> -->
</launch>
